ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityRewardedAd.java

```diff
@@ -0,0 +1,272 @@
+                                                               unityAdsInitializationError, String s) {
+                        Log.e(UnityAdapter.TAG, "Unity Ads initialization failed: [" +
+                                unityAdsInitializationError + "] " + s + ", cannot load rewarded ad"
+                                + "for placement ID '" + getPlacementId() + "' in game '" + gameId + "'");
+                    }
+                });
+    }
+
+    /**
+     * This method will load Unity Ads for a given Placement ID and send the ad loaded event if the
+     * ads have already loaded.
+     * @param placementId   Placement ID of the ad to be loaded
+     */
+    protected void loadRewardedAd(String placementId) {
+
+        UnityAds.load(placementId, mUnityLoadListener);
+
+    }
+
+    @Override
+    public void showAd(Context context) {
+
+        if (!(context instanceof Activity)) {
+            String message = "Unity Ads failed to show rewarded ad: activity context is required.";
+            Log.w(UnityAdapter.TAG, message);
+            if (mMediationRewardedAdCallback != null) {
+                mMediationRewardedAdCallback.onAdFailedToShow(message);
+            }
+            return;
+        }
+        Activity activity = (Activity) context;
+
+        // Check if the placement is ready before showing
+        if (UnityAds.isReady(getPlacementId())) {
+
+            // Every call to UnityAds#show will result in an onUnityAdsFinish callback (even when
+            // Unity Ads fails to show an ad).
+
+            UnityAds.show(activity, getPlacementId());
+
+            // Unity Ads does not have an ad opened callback.
+            if (mMediationRewardedAdCallback != null) {
+                mMediationRewardedAdCallback.onAdOpened();
+                mMediationRewardedAdCallback.reportAdImpression();
+            }
+        } else {
+            String message = "Unity Ads failed to show rewarded ad. Placement ID '" +
+                    getPlacementId() + "' is not ready.";
+            Log.w(UnityAdapter.TAG, message);
+            if (mMediationRewardedAdCallback != null) {
+                mMediationRewardedAdCallback.onAdFailedToShow(message);
+            }
+        }
+
+    }
+
+    /**
+     * Checks whether or not the provided Unity Ads IDs are valid.
+     *
+     * @param gameId      Unity Ads Game ID to be verified.
+     * @param placementId Unity Ads Placement ID to be verified.
+     * @return {@code true} if all the IDs provided are valid.
+     */
+    private static boolean isValidIds(String gameId, String placementId) {
+        if (TextUtils.isEmpty(gameId) || TextUtils.isEmpty(placementId)) {
+            String ids = TextUtils.isEmpty(gameId) ? TextUtils.isEmpty(placementId)
+                    ? "Game ID and Placement ID" : "Game ID" : "Placement ID";
+            Log.w(UnityAdapter.TAG, ids + " cannot be empty.");
+
+            return false;
+        }
+        return true;
+    }
```

Comment:
This method defined in multiple places...

Reply:
I will change to public and put it just in UnityAdapter